modules/service.sh

```diff
@@ -0,0 +1,52 @@
+# shellcheck disable=SC2039
+
+check_user() {
+    if [ "$(id -u)" -ne 0 ]; then
+        error_msg "This command must be run as root (try using sudo)"
+        exit 2
+    fi
+}
+
+service_from_command() {
+    local command="$1"
+
+     echo "$command" | awk '
+         /^is-.*-enabled$/ {
+             match($0, /^is-(.*)-enabled$/, m);
+             print m[1];
+         }
+         /^(enable|disable)-/ {
+             match($0, /^(enable|disable)-(.*)/, m);
+             print m[2];
+         }
+    '
+}
+
+service_enable() {
+    local service="$1"
+    local token="$2"
+
+    check_user
+    service_check_support "$service"
+    "${service}_validate_token" "$token" || exit 3
+    "${service}_enable" "$token"
+}
+
+service_is_enabled() {
+    local service="$1"
+
+    "${service}_is_enabled"
+}
+
+service_check_support() {
```

Comment:
We have check_service_support() and service_check_support() now, this is getting confusing.

Reply:
Maybe it's time to better distinguish really internal functions from functions a module writer should use. Use a "_" prefix maybe. That being said, this confusion between check_service_support and service_check_support can only be sorted with a new name I think.

Reply:
I agree, `check_service_support` is badly named, I'm not sure if it's better, but I'm renaming it to `check_support` (which matches `check_user`).

The idea is to have functions from each module to use the module name prefix (hence `service_`, `esm_`, `apt_`, ... ) but some generic functions don't really follow that naming.

I've been using the `_` prefix for functions that are only meant for use inside the file where they're declared.

Hopefully the rename above should be enough to make the distinction clearer